src/machine/machined-varlink.c

```diff
@@ -388,14 +388,66 @@ static int vl_method_get_memberships(sd_varlink *link, sd_json_variant *paramete
         return sd_varlink_error(link, "io.systemd.UserDatabase.NoRecordFound", NULL);
 }
 
-static int list_machine_one(sd_varlink *link, Machine *m, bool more) {
-        int r;
+typedef enum ListMachineFlags {
+        LIST_MACHINE_NEED_MODE          = 1 << 0,
+        LIST_MACHINE_ACQUIRE_ADDRESSES  = 1 << 1,
+        LIST_MACHINE_ACQUIRE_OS_RELEASE = 1 << 2,
+        LIST_MACHINE_ACQUIRE_UID_SHIFT  = 1 << 3,
+} ListMachineFlags;
+
+static int list_machine_one_and_maybe_read_metadata(sd_varlink *link, Machine *m, ListMachineFlags flags) {
+        _cleanup_free_ struct local_address *addresses = NULL;
+        _cleanup_strv_free_ char **os_release = NULL;
+        uid_t shift = UID_INVALID;
+        int n, r;
 
         assert(link);
         assert(m);
 
         _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
 
+        /* TODO(ikruglov): maybe swallow errors and don't output fields if NoPrivateNetwork/NoOSReleaseInformation or NotSupported?
+         * I think swallowing errors in general listing (varlinkctl --more call ... io.systemd.Machine.List '{}') make sense.
+         * However, when a user request information about specific machine (varlinkctl --more call ... io.systemd.Machine.List '{"name": "foo"}')
+         * giving a error is better */
```

Comment:
maybe let's make this explicit, by requesting the metadata to be acquired via a an enum (and make it a single one maybe for all three kind of metadata).

i.e. acquireMachineMetadata as one of "yes", "no", "graceful". The difference between "yes" and "graceful" would be that the latter eats up errors

Reply:
'graceful' mode make sense to me. I'll implement it.

Reply:
making it a single knob would make it nicely symmetric to the image list API you added which has a single knob.

Reply:
(btw, maybe convert the image list thing to a similar concept of "graceful" metadata acquiring? at least in the "public" interface)